dev/cuda/adamw.cu

```diff
@@ -159,8 +159,8 @@ int main(int argc, char **argv) {
     // create random data on host (to be used for the CPU reference implementation)
     float* params_memory = make_random_float(num_parameters);
     float* grads_memory = make_random_float(num_parameters);
-    float* m_memory = make_random_float_01(num_parameters);
-    float* v_memory = make_random_float_01(num_parameters);
+    float* m_memory = make_random_float(num_parameters);
+    float* v_memory = make_random_float(num_parameters);
```

Comment:
I think v_memory should be >= 0, as its is mean squared gradient?

Reply:
Correct!

On a side note: ideally we'd have Gaussian distribution supported and not only uniform in a certain range.